src/import_edf.jl

```diff
@@ -204,22 +222,49 @@ can be either a `String` giving the generated channel name, or a `Pair` mapping
 a canonical name to a list of alternatives that it should be substituted for
 (e.g., `"canonical_name" => ["alt1", "alt2", ...]`).
 
+`unit_alternatives` lists standardized unit names and alternatives that map to them.
+See `OndaEDF.STANDAR_UNITS` for defaults.
+
+`preprocess_labels(label::String, transducer_type::String)` is applied to raw edf signal header labels
+beforehand; defaults to returning `label`.
+
 See `OndaEDF.STANDARD_LABELS` for the labels (`signal_names => channel_names`
 `Pair`s) that are used to extract EDF signals by default.
 
 """
-function extract_channels_by_label(edf::EDF.File, signal_names, channel_names)
+function extract_channels_by_label(edf::EDF.File, signal_names, channel_names; unit_alternatives=STANDARD_UNITS, preprocess_labels=(l,t) -> l)
     matcher = x -> begin
         # yo I heard you like closures
-        return s -> match_edf_label(s.header.label, signal_names,
-                                    x isa Pair ? first(x) : x,
+        # x is either a channel name (string), or a channel_name => alternatives Pair.
+        this_channel_name = x isa Pair ? first(x) : x
+        return s -> match_edf_label(preprocess_labels(s.header.label, s.header.transducer_type),
+                                    signal_names,
+                                    this_channel_name,
                                     channel_names)
     end
-    edf_channel_names, edf_channels = extract_channels(edf.signals, (matcher(x) for x in channel_names))
-    isempty(edf_channel_names) && return nothing
+    edf_channels = extract_channels(edf.signals, (matcher(x) for x in channel_names))
+
+    grouped = groupby(edf_channels) do p
+        channel_name, edf_signal = p
+        return ((edf_signal.header.samples_per_record / edf.header.seconds_per_record), edf_signal.header.physical_dimension)
+    end
 
-    info = edf_signals_to_samplesinfo(edf, edf_channels, first(signal_names), edf_channel_names)
-    return info, edf_channels
+    results = map(values(grouped)) do pairs
+        edf_channel_names, edf_channels = zip(pairs...)
+        try
+            edf_channels = collect(edf_channels)
+            info = edf_signals_to_samplesinfo(edf, edf_channels, first(signal_names), collect(edf_channel_names))
+            return info, edf_channels
+        catch e
+            units = [s.header.label => s.header.physical_dimension for s in edf_channels]
+            msg ="""Skipping signal: error while processing units and encodings
+                    for $(first(signal_names)) signal with units $units"""
+            return SamplesInfoError(msg, e)
+        end
+    end
```

Comment:
I'm having a hard time following what's going on.  Either more informative variable names (what are the pairs? what is grouped?) or some comments would help a lot I think.

Reply:
I pushed some comments, hopefully that helps.

I also added a bullet point in the main comment above explaining the grouping.